reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,304 @@
+            regions.
+        """
+        logger.info('Loading all files')
+        with Geotiff(iso_regions_f) as gt:
+            self._iso_regions = gt.values[0]
+            self._profile = gt.profile
+
+        with ExclusionLayers(excl_h5) as el:
+            self._slope = el[slope_lyr]
+            self._land_use = el[nlcd_lyr]
+            # self._write_geotiff('out/slope.tif', self._profile, self._slope)
+            # self._write_geotiff('out/land_use.tif', self._profile, self._land_use)
+
+        assert self._iso_regions.shape == self._land_use.shape == \
+            self._slope.shape, 'All rasters must be the same shape'
+        logger.info('Loading files complete')
+
+        self._profile['dtype'] = 'float32'
+        self._profile['nodata'] = None
+
+        self._land_use_classes = land_use_classes
+        self._iso_mults = iso_mults
+        self._base_line_costs = base_line_costs
+        self._iso_lookup = iso_lookup
+        self._default_mults = default_mults
+        self._cell_size = CELL_SIZE
+
+    @classmethod
+    def build_cost_rasters(cls, iso_regions_f,
+                           h5_f='costs.h5',
+                           land_use_classes=NLCD_LAND_USE_CLASSES,
+                           iso_mults_f=None, base_line_costs_f=None,
+                           iso_lookup_f=None, power_classes_f=None,
+                           default_mults={}, save_to_h5=True):
+        """
+        Build cost rasters using base line costs and multipliers. Save to
+        h5 file or tiff
+
+        Parameters
+        ----------
+        iso_regions_f : str
+            File with raster of ISO regions
+        h5_f : str
+            H5 file to save costs to
+        land_use_classes : dict
+            NCLD land use codes corresponding to use classes for multipliers
+        iso_mults_f : None | str
+            ISO multipliers JSON file. Use default if None
+        base_line_costs_f : None | str
+            Base line cost per mile for ISOs file. Use default if None
+        iso_lookup_f : None | str
+            JSON file linking ISO region code to name. Use default if None
+        power_classes_f : None | str
+            JSON of line capacities to use for reV power classes. Use default
+            if None
+        default_mults : dict
+            Multipliers for regions not specified in iso_mults_f.
+        save_to_h5 : bool
+            Save output to h5 if true, save to geotiff if false
+        """
+        xc = XmissionConfig(iso_mults_f=iso_mults_f,
+                            base_line_costs_f=base_line_costs_f,
+                            iso_lookup_f=iso_lookup_f,
+                            power_classes_f=power_classes_f)
+
+        xcc = cls(iso_regions_f, land_use_classes, xc.iso_mults,
+                  xc.base_line_costs, xc.iso_lookup, default_mults)
+        mults_arr = xcc.create_multipliers()
+
+        if save_to_h5:
+            xcc._init_h5(h5_f, iso_regions_f)
```

Comment:
Move to class __init__

Reply:
done, need to use self._profile instead of iso_regions_f to init

Reply:
See new comment, I think this is right